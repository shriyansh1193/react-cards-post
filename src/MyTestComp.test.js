import React from 'react';
import MyTestComp from './MyTestComponent';
import { create } from 'react-test-renderer';

describe('Our First Snapshot Test Case', () => {
    test('Testing the Button UI for the App', () => {
        let main = create(<MyTestComp />)
        expect(main.toJSON()).toMatchSnapshot();
    })
});


// describe('Changing our Button Name to Hide', () => {
//     test('toggle the button', () => {
//         let tree = create(<MyTestComp />);
//         let instance = tree.getInstance();

//         expect(instance.state.isActive).toBe(false)

//         // Change the State
//         instance.handleClick();

//         // isActive Property is Updated to 'true'
//         expect(instance.state.isActive).toBe(true);

//         expect(tree.toJSON()).toMatchSnapshot();    
//     })
// });

test('toggle the button', () => {
    let tree = create(<MyTestComp />);
    let instance = tree.getInstance();

    expect(instance.state.isActive).toBe(false)

    // Change the State
    instance.handleClick();

    // isActive Property is Updated to 'true'
    expect(instance.state.isActive).toBe(true);

    expect(tree.toJSON()).toMatchSnapshot();    
});

describe('Checking our Raining Status', () => {
    test('Check the Rain', () => {
        let tree = create(<MyTestComp />);
        let instance = tree.getInstance();

        expect(instance.state.isRaining).toBe(false)

        // Change the State
        instance.checkRain();

        // isRaining Property is Checked to 'rain' or 'not raining'
        expect(instance.state.isRaining).toBe(true);

        expect(tree.toJSON()).toMatchSnapshot();
    })
});