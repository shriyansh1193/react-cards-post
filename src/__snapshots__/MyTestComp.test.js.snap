// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking our Raining Status Check the Rain 1`] = `
<div
  className="MyTestComp"
>
  <button
    onClick={[Function]}
  >
    Show
  </button>
  <button
    onClick={[Function]}
  >
    Not Raining
  </button>
</div>
`;

exports[`Our First Snapshot Test Case Testing the Button UI for the App 1`] = `
<div
  className="MyTestComp"
>
  <button
    onClick={[Function]}
  >
    Show
  </button>
  <button
    onClick={[Function]}
  >
    Not Raining
  </button>
</div>
`;

exports[`toggle the button 1`] = `
<div
  className="MyTestComp"
>
  <button
    onClick={[Function]}
  >
    Show
  </button>
  <button
    onClick={[Function]}
  >
    Not Raining
  </button>
</div>
`;
